### ISSUE TITLE
mypy false negative when trying to access an instance variable as if it were a class variable

### ISSUE THREAD

**Issue by FeldrinH** [python/mypy#15270]
**Bug Report**

mypy finds no issues when the code tries to access an instance variable as if it were a class variable.

**To Reproduce**

The following code produces an error at runtime when accessing `SimpleClass.a`, but `mypy --strict` finds no issues.
```python
class SimpleClass:
    def __init__(self) -> None:
        self.a = "test"

print(SimpleClass.a)
```

**Expected Behavior**

I would have expected accessing `SimpleClass.a` to be flagged as an error, given that `a` is clearly an instance variable.

**Actual Behavior**

`mypy --strict` found no issues.

**Your Environment**

- Mypy version used: mypy 1.3.0
- Mypy command-line flags: `--strict`
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.10.11

**Comment by Mr-Pepe** [comment]
Duplicate of https://github.com/python/mypy/issues/240
